src/concerns/autoplay.coffee

```diff
@@ -0,0 +1,24 @@
+###
+Code related to auotplay features of the carousel
+###
+export default
+	props:
+		# A delay provided in seconds for the autoplay. 0 is disabled
+		autoplayDelay:
+			type: Number
+			default: 0
+
+		pauseOnFocus:
+			type: Boolean
+			default: true
+
+	computed:
+		# Stop animation if window is hidden or if carousel is focused
+		paused: -> (@windowHidden or @isFocused) if @pauseOnFocus
+
+	mounted: ->
+		if @autoplayDelay then @autoPlayInterval = setInterval (() =>
+			@next() if not @paused
+			), @autoplayDelay * 1000
```

Comment:
I think this should be refactored to be `setTimeout` based.  I've run into issues in the past where browsers will queue up setInterval callbacks and then execute them rapidly in a series.  Like if you switch browser tabs.  Maybe the `visibilitychange` watching would prevent this, but just to play it safe, I recommend changing to something like:

```coffee
mounted: -> @autoplayStart() if @autoplayDelay
beforeDestroy: -> @autoplayStop()

methods:

  autoplayStart: -> @autoplayId = setTimeout @autoplayNext, @autoplayDelay * 1000

  autoplayNext: ->
    @next()
    @autoplayStart()
    
   autoplayStop: -> clearTimeout @autoplayId
```

Reply:
I haven't hit any issues with any queueing at this point with the `visibilitychange` added in (thought that was the point of adding in that check, but happy to swap it to `setTimeout` if requested.

Reply:
Ok, well let's trust in `visibilitychange` then. If we need to, we can refactor it later and it wouldn't be a breaking change.